File: 3195-separate-black-and-white-balls/3195-separate-black-and-white-balls.java
class Solution {
    public long minimumSteps(String s) {
        long black = 0;
        long minSwaps = 0;
        int n = s.length();
        for(int i = 0 ; i < n; i++) {
            if(s.charAt(i) == '1') black++;
            else minSwaps += black;
        }
        return minSwaps;
    }
}

File: 3195-separate-black-and-white-balls/3195-separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        long long res=0;
        int n=s.size();
        int i=0,j=n-1;

        while(i<j){

            if(s[i]=='1'){
                if(s[j]=='0'){
                    res+=j-i;
                    i++;
                }
                j--;
            }
            else{
                if(s[j]=='1'){
                    j--;
                }
                i++;
            }

        }
        return res;
    }
};